src/flow/domain/mod.rs

```diff
@@ -937,6 +997,26 @@ impl Domain {
                         drop(n);
 
                         if let Packet::None = m {
+                            if !target && partial_key.is_some() && is_transactional {
```

Comment:
"The partial replay was captured, but we still need to propagate an (ignored) `ReplayPiece` so that downstream domains don't end up waiting forever for the timestamp we claimed."